src/gui/torrentoptionsdialog.cpp

```diff
@@ -0,0 +1,342 @@
+    const int globalDownloadLimit = isAltLimitEnabled
+            ? (BitTorrent::Session::instance()->altGlobalDownloadSpeedLimit() / 1024)
+            : (BitTorrent::Session::instance()->globalDownloadSpeedLimit() / 1024);
+
+    const int uploadVal = qMax(0, (oldUploadLimit / 1024));
+    const int downloadVal = qMax(0, (oldDownloadLimit / 1024));
+    int maxUpload = (globalUploadLimit <= 0) ? 10000 : globalUploadLimit;
+    int maxDownload = (globalDownloadLimit <= 0) ? 10000 : globalDownloadLimit;
+
+    // This can happen for example if global rate limit is lower than torrent rate limit.
+    if (uploadVal > maxUpload)
+        maxUpload = uploadVal;
+    if (downloadVal > maxDownload)
+        maxDownload = downloadVal;
+
+    m_ui->sliderUploadLimit->setMaximum(maxUpload);
+    m_ui->sliderUploadLimit->setValue(uploadVal);
+    m_ui->sliderDownloadLimit->setMaximum(maxDownload);
+    m_ui->sliderDownloadLimit->setValue(downloadVal);
+    updateUploadSpinValue(uploadVal);
+    updateDownloadSpinValue(downloadVal);
+
+    bool useGlobalValue = false;
+    if (allSameShareLimits)
+    {
+        useGlobalValue = (torrents[0]->ratioLimit() == BitTorrent::TorrentHandle::USE_GLOBAL_RATIO)
+                && (torrents[0]->seedingTimeLimit() == BitTorrent::TorrentHandle::USE_GLOBAL_SEEDING_TIME);
+    }
+    if (useGlobalValue || !allSameShareLimits)
+    {
+        m_ui->radioUseGlobalShareLimits->setChecked(true);
+    }
+    else if ((currentMaxRatio == BitTorrent::TorrentHandle::NO_RATIO_LIMIT)
+             && (currentMaxSeedingTime == BitTorrent::TorrentHandle::NO_SEEDING_TIME_LIMIT))
+    {
+        m_ui->radioNoLimit->setChecked(true);
+        currentMaxRatio = session->globalMaxRatio();
+        currentMaxSeedingTime = session->globalMaxSeedingMinutes();
+    }
+    else
+    {
+        m_ui->radioTorrentLimit->setChecked(true);
+        if (currentMaxRatio >= 0)
+            m_ui->checkMaxRatio->setChecked(true);
+        if (currentMaxSeedingTime >= 0)
+            m_ui->checkMaxTime->setChecked(true);
+    }
+    m_ui->ratioSpinBox->setValue(currentMaxRatio);
+    m_ui->timeSpinBox->setValue(currentMaxSeedingTime);
+
+    if (!allTorrentsArePrivate)
+    {
+        if (m_ui->checkDisableDHT->checkState() != Qt::PartiallyChecked)
+            m_ui->checkDisableDHT->setChecked(disableDHT);
+        if (m_ui->checkDisablePEX->checkState() != Qt::PartiallyChecked)
+            m_ui->checkDisablePEX->setChecked(disablePEX);
+        if (m_ui->checkDisableLSD->checkState() != Qt::PartiallyChecked)
+            m_ui->checkDisableLSD->setChecked(disableLSD);
+    }
+    else
+    {
+        const QString privateTorrentsTooltip = tr("Selected torrents are private");
+        m_ui->checkDisableDHT->setToolTip(privateTorrentsTooltip);
+        m_ui->checkDisablePEX->setToolTip(privateTorrentsTooltip);
+        m_ui->checkDisableLSD->setToolTip(privateTorrentsTooltip);
+        m_ui->checkDisableDHT->setChecked(false);
```

Comment:
Shouldn't they be checked+disabled for private torrents?

Reply:
And provide the illusion that these settings are applied to private torrents? But on the other hand, checked will be more intuitive for users i guess. Will change.